cron.php fails because there is no check whether ElasticSearch is enabled

#### Issue
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->
OS: Linux
SuiteCRM: 7.11.2 (Fresh install)
On a shared host (I am unable to configure ElasticSearc as per the manual)

Check also Forum thread:
https://suitecrm.com/suitecrm/forum/feedback/21947-cron-jobs-failure-because-of-elasticsearch-which-is-disabled#78465

I am getting an error in the SuiteCRM log every minute when cron.php is launched by the scheduler.
There is no error in the PHP/Apache log.

This failure stops the execution of all Scheduler Jobs so therefore it is a big problem!

This is the error that appears in the log:
`Tue Feb 26 14:16:01 2019 [2703158][1][FATAL] [ERROR] Failed to index bean to index
Tue Feb 26 14:16:01 2019 [2703158][1][FATAL] [ERROR] SuiteCRM\Search\Exceptions\SearchException: Elasticsearch trying to re-indexing a bean but indexer is disabled in configuration. in /var/www/vhosts/website.com/httpdocs/crm.website.net/suitecrm/lib/Search/ElasticSearch/ElasticSearchHooks.php:121
Stack trace:
#0 /var/www/vhosts/website.com/httpdocs/crm.website.net/suitecrm/lib/Search/ElasticSearch/ElasticSearchHooks.php(103): SuiteCRM\Search\ElasticSearch\ElasticSearchHooks->reIndex(Object(SchedulersJob))
#1 /var/www/vhosts/website.com/httpdocs/crm.website.net/suitecrm/lib/Search/ElasticSearch/ElasticSearchHooks.php(76): SuiteCRM\Search\ElasticSearch\ElasticSearchHooks->reIndexSafe(Object(SchedulersJob))
#2 /var/www/vhosts/website.com/httpdocs/crm.website.net/suitecrm/include/utils/LogicHook.php(272): SuiteCRM\Search\ElasticSearch\ElasticSearchHooks->beanSaved(Object(SchedulersJob), 'after_save', '')
#3 /var/www/vhosts/website.com/httpdocs/crm.website.net/suitecrm/include/utils/LogicHook.php(213): LogicHook->process_hooks(Array, 'after_save', '')
#4 /var/www/vhosts/website.com/httpdocs/crm.website.net/suitecrm/data/SugarBean.php(3129): LogicHook->call_custom_logic('SchedulersJobs', 'after_save', '')
#5 /var/www/vhosts/website.com/httpdocs/crm.website.net/suitecrm/data/SugarBean.php(2442): SugarBean->call_custom_logic('after_save', '')
#6 /var/www/vhosts/website.com/httpdocs/crm.website.net/suitecrm/modules/SchedulersJobs/SchedulersJob.php(320): SugarBean->save()
#7 /var/www/vhosts/website.com/httpdocs/crm.website.net/suitecrm/modules/SchedulersJobs/SchedulersJob.php(546): SchedulersJob->resolveJob('success')
#8 /var/www/vhosts/website.com/httpdocs/crm.website.net/suitecrm/include/SugarQueue/SugarCronJobs.php(191): SchedulersJob->runJob()
#9 /var/www/vhosts/website.com/httpdocs/crm.website.net/suitecrm/include/SugarQueue/SugarCronJobs.php(231): SugarCronJobs->executeJob(Object(SchedulersJob))
#10 /var/www/vhosts/website.com/httpdocs/crm.website.net/suitecrm/cron.php(99): SugarCronJobs->runCycle()
#11 {main}
`

#### Expected Behavior
<!--- Tell us what should happen -->
Somewhere in the code, instead of just executing the ElasticSearch indexing, there should be a check to find out whether ElasticSearch has been enabled and only if this condition is true, the indexing should take place.

#### Actual Behavior
<!--- Tell us what happens instead -->
The ElasticSearch indexing is executed at every cron.php instance even when ElasticSearch is not enabled. This leads to a FATAL error (see log above in the description).

#### Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->
Modify the following file:
lib/Search/ElasticSearch/ElasticSearchHooks.php

Old/current code:
`/**
     * Callback for save beans.
     *
     * @param SugarBean $bean
     * @param           $event
     * @param           $arguments
     */
    public function beanSaved(SugarBean $bean, $event, $arguments)
    {
        $this->action = 'index';

        $this->reIndexSafe($bean);
    }`

Code after modification:
`/**
     * Callback for save beans.
     *
     * @param SugarBean $bean
     * @param           $event
     * @param           $arguments
     */
    public function beanSaved(SugarBean $bean, $event, $arguments)
    {
        $this->action = 'index';

        if (ElasticSearchIndexer::isEnabled() !== false)
          $this->reIndexSafe($bean);
    }
`

(I have only added one line:
`        if (ElasticSearchIndexer::isEnabled() !== false)
`
and indented the next one)

#### Steps to Reproduce
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug include code to reproduce, if relevant -->
1. Set up a crontab entry for SuiteCRM on versions 7.11.0/1/2
2. Make sure that in Admin ElasticSearch is disabled
3. Check the logs to see the error at every run of cron.php
4. 

#### Context
<!--- How has this bug affected you? What were you trying to accomplish? -->
<!--- If you feel this should be a low/medium/high priority then please state so -->
Medium priority: the scheduled jobs after the error may not be executed

#### Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* SuiteCRM Version used: versions 7.11.0/1/2
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): Chrome latest version
* Environment name and version (e.g. MySQL, PHP 7): MySQL + PHP 7.1
* Operating System and version (e.g Ubuntu 16.04): Linux CentOS

--- a/lib/Search/ElasticSearch/ElasticSearchHooks.php
+++ b/lib/Search/ElasticSearch/ElasticSearchHooks.php
@@ -73,7 +73,9 @@ class ElasticSearchHooks
     {
         $this->action = 'index';
 
-        $this->reIndexSafe($bean);
+        if (ElasticSearchIndexer::isEnabled() !== false) {
+            $this->reIndexSafe($bean);
+        }
     }
 
     /**

